conda_index/index/__init__.py

```diff
@@ -712,17 +721,17 @@ def index(
 
                         t2.set_description("Writing patched repodata")
                         t2.update()
-                        log.debug("write patched repodata")
+                        log.debug("%s write patched repodata", subdir)
                         self._write_repodata(subdir, patched_repodata, REPODATA_JSON_FN)
                         t2.set_description("Building current_repodata subset")
                         t2.update()
-                        log.debug("build current_repodata")
+                        log.debug("%s build current_repodata", subdir)
                         current_repodata = _build_current_repodata(
                             subdir, patched_repodata, pins=current_index_versions
                         )
                         t2.set_description("Writing current_repodata subset")
                         t2.update()
-                        log.debug("write current_repodata")
+                        log.debug("%s write current_repodata", subdir)
```

Comment:
the write repodata steps could be run in a subprocess but they may be RAM-limited, since they have to hold the entire repodata.json in memory